### ISSUE TITLE
False positive for `new_without_default_derive`

### ISSUE THREAD

**Issue by sgrif** [rust-lang/rust-clippy#3632]
Given the following code:

```rust
use std::collections::HashMap;

#[derive(Default)]
#[allow(missing_debug_implementations)] // Can't derive debug
pub struct Registry<Env> {
    job_types: HashMap<&'static str, Env>,
}

impl<Env> Registry<Env> {
    pub fn new() -> Self {
        Registry {
            job_types: Default::default(),
        }
    }
}
```

Clippy will suggest adding `#[derive(Default)]` on a struct that already has it.

Rust 1.31.1
clippy 0.0.212 (2e26fdc2 2018-11-22)

**Comment by mikerite** [comment]
I can't reproduce this error on the playground. 
The playground's current version 0.0.212 (2018-12-30 39bd844).
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7539c39de52057b7e0f0fc5a5d291f29)

**Comment by flip1995** [comment]
Maybe this was fixed in #3558.
